tests/helpers/common/Chocolatey/Invoke-Choco.ps1

```diff
@@ -33,28 +33,43 @@
 
             $remainingArguments
         )
-
     }
     end {
+        $stopwatch = [System.Diagnostics.Stopwatch]::new()
+        $stopwatch.Start()
+
         $output = if ($PipelineInput) {
             $PipelineInput | & $chocoPath @arguments
         }
         else {
             & $chocoPath @arguments
         }
 
+        $stopwatch.Stop()
+        # We do not use env:ChocolateyInstall here as it is not guaranteed due to snapshots.
+        # Saving into the default Chocolatey install location as .log so Test Kitchen can pick up the file.
+        [PSCustomObject]@{
+            Date       = Get-Date -Format 'o'
+            Duration   = $stopwatch.Elapsed
+            Invocation = ($MyInvocation.PositionMessage -split "`r`n")[0]
+            Arguments  = ($Arguments -join ' ') -replace "`r`n", ''
+            ExitCode   = $LastExitCode
+        } | Export-Csv -Path $env:ALLUSERSPROFILE/chocolatey/logs/testInvocations.log -NoTypeInformation -Append
+
         [PSCustomObject]@{
             # We trim all the lines, so we do not take into account
             # trimming the lines when asserting, and that extra whitespace
             # is not considered in our assertions.
-            Lines    = if ($output) {
+            Lines     = if ($output) {
                 $output.Trim()
             }
             else {
                 @()
             }
-            String   = $output -join "`r`n"
-            ExitCode = $LastExitCode
+            String    = $output -join "`r`n"
+            ExitCode  = $LastExitCode

```

Comment:
Since we are doing custom logic between the time we launch any application, and using the `LastExitCode`, then lets store it in a separate variable right after the application launch to make sure we don't add/modify anything that could result in the exit code being overwritten.

Reply:
That occurred to me after I initially added this. I did some tests that showed that the variable didn't get overridden, at least not with what's currently there. But I think you're right that we should store it so it doesn't accidentally get clobbered.

Reply:
I've updated this to store the exit code after we get it and returning that.